gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java

```diff
@@ -490,48 +465,56 @@ public Single<Certificate> rotate(String domain, User principal) {
                         .referenceType(ReferenceType.DOMAIN)
                         .type(EventType.CERTIFICATE_CREATED).throwable(error)));
     }
-
+    private Date checkCertificateExireDate(String type, String storePass, String alias, byte[] cert) throws KeyStoreException, IOException, NoSuchAlgorithmException, CertificateException {
+        var store = switch(type) {
+            case JAVA_KEYSTORE_CERTIFICATE -> KeyStore.getInstance("JKS");
+            case PKCS_12_CERTIFICATE -> KeyStore.getInstance("PKCS12");
+            default -> throw new CertificateException("Unsupported certification type: " + type);
+        };
+        store.load(new ByteArrayInputStream(cert), storePass.toCharArray());
+        var expiryDate = ((X509Certificate) store.getCertificate(alias)).getNotAfter();
+        if (Instant.now().isAfter(expiryDate.toInstant())) {
+            throw new CertificateException("Uploading certificate is already expired");
+        }
+        store.deleteEntry(alias);
+        return expiryDate;
+    }
```

Comment:
**TODO:** I don't think we have to reimplement the KeyStore loading here. We should instead provide a "isValid" method into the CertificateManager that will create a CertificateProvider and then reject the cert if the expiration date is not valid. https://github.com/gravitee-io/gravitee-access-management/blob/833fb5e34162991f5d30fd43d9a198237d24ecae/gravitee-am-management-api/gravitee-am-management-api-service/src/main/java/io/gravitee/am/management/service/impl/CertificateManagerImpl.java#L152

We can't use the loadCertificate as it is but we can extract the certificate provider creation to use the plugin instead of reimplementing the logic. In that way, if we introduce a new certificate plugin type, we will not forget to update this switch
```
            var providerConfig = new CertificateProviderConfiguration(certificate);
            var certificateProvider = certificatePluginManager.create(providerConfig);
```